src/Quartz.Tests.Unit/SchedulerTest.cs

```diff
@@ -303,16 +303,19 @@ public async Task ReschedulingTriggerShouldKeepOriginalNextFireTime()
             await scheduler.Start();
 
             var job = JobBuilder.Create<NoOpJob>().Build();
+            var startAtOffset = TimeSpan.FromSeconds(1);  //Has to be enough time for the initial assert to verify job has not yet run.
             IOperableTrigger trigger = (IOperableTrigger) TriggerBuilder.Create()
                 .WithSimpleSchedule(x => x.WithIntervalInHours(1).RepeatForever())
                 .ForJob(job)
-                .StartNow()
+                .StartAt(DateTime.UtcNow.Add(startAtOffset))
                 .Build();
 
             await scheduler.ScheduleJob(job, trigger);
-
             trigger = (IOperableTrigger) await scheduler.GetTrigger(trigger.Key);
             Assert.That(trigger.GetPreviousFireTimeUtc(), Is.EqualTo(null));
+            
+            // Ensure trigger will fire by waiting for elapsed StartAt time.
+            Thread.Sleep(Convert.ToInt16(startAtOffset.TotalMilliseconds));
```

Comment:
`await Task.Delay(startAtOffset);` would read better, but I also see that this has already been improved in `main`, maybe should just port the fixes from there?

Reply:
sure, didn't even think to check main... will refresh.

Reply:
Done, backported main test to 3.x